src/java.desktop/share/classes/javax/swing/plaf/metal/MetalBorders.java

```diff
@@ -217,16 +239,27 @@ public Insets getBorderInsets(Component c, Insets newInsets) {
      */
     @SuppressWarnings("serial") // Superclass is not serializable across versions
     public static class InternalFrameBorder extends AbstractBorder implements UIResource {
-        private static final int corner = 14;
+        private static int corner = 14;
```

Comment:
Before it was final, now it is not and it is static - means you are changing it down the line. What will happen if two instances will try to change it in a different ways - i.e. if there are more than one window and they are on a screens with different scale factors? I do not really get the logic - either this value is a local one and being calculated for each instance or it is a final and constant for all. And seeing how you update it continuously - will it not accumulate this change is paintBorder will be called multiple times?

Reply:
@azuev-java Thank you for reviewing and bringing up - **_"multiple windows on multiple screens with different scaling"_** scenario. I missed accounting for the above scenario in the present fix. Yes, I believe that `corner` should be local variable that is calculated depending on the screen's scale on which it is displayed. 
I was also thinking if we can may be precompute & store all the corner values in a hashmap for all the scale values that we can expect?

Reply:
> I was also thinking if we can - may be precompute & store all the corner values in a `HashMap` for all the scale values that we can expect?

It depends on how often this value is being computed and how complicated the computation is. Since we request the current transform value anyways on every paint i do not see how accessing the HashMap would be more beneficial than the simple floating point multiplication and a rounding up the result. Just make the static final constant storing the default corner size and a local variable that stores the calculated current corner. You can try to use HashMap but i doubt it will make much difference performance wise. It is not a complex calculation. I might be wrong so you can try both ways and check for the performance impact.